### PR TITLE
Admin category page : create - edit -delete

### PR DESCRIPTION
**Pull request by truongnv0211**

### CODE REVIEW

File: app/controllers/admin/categories_controller.rb [hunk 1]
```diff
@@ -0,0 +1,48 @@
+class Admin::CategoriesController < ApplicationController
+  before_action :required_admin
+  before_action :set_category, only: [:show, :edit, :update, :destroy]
```
Comment (phamkykhoi) [inline app/controllers/admin/categories_controller.rb]: not use only.
should use except

File: app/controllers/admin/categories_controller.rb [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+  def index
+    @categories = Category.paginate page: params[:page], per_page: 3
```
Comment (phamkykhoi) [inline app/controllers/admin/categories_controller.rb]: Settings

File: app/controllers/admin/categories_controller.rb [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+  end
```
Comment (phamkykhoi) [inline app/controllers/admin/categories_controller.rb]: Break line

File: app/models/word.rb [hunk 4]
```diff
@@ -5,4 +5,6 @@ class Word < ActiveRecord::Base
 
   validates :jp_word, presence: true
   validates :correct_answer, presence: true, numericality: {only_integer: true}
+
+  scope :located_in, ->category{where category_id: category.id}
 end
```
Comment (phamkykhoi) [inline app/models/word.rb]: Remove has_many :lessons, through: :lesson_words

File: app/controllers/admin/categories_controller.rb [hunk 5]
```diff
@@ -0,0 +1,48 @@
+class Admin::CategoriesController < ApplicationController
+  before_action :required_admin
+  before_action :set_category, except: [:index, :new, :create]
+
+  def index
+    @categories = Category.paginate page: params[:page], per_page: Settings.per_page
+  end
+  def new
```
Comment (phamkykhoi) [inline app/controllers/admin/categories_controller.rb]: Insert line empty above, please.

File: app/views/admin/categories/_category.html.erb [hunk 6]
```diff
@@ -0,0 +1,6 @@
+<h3><%= category.title %></h3>
+<i><%= Word.located_in(category).count %> - words</i>
```
Comment (phamkykhoi) [inline app/views/admin/categories/_category.html.erb]: I18n

File: app/controllers/sessions_controller.rb [hunk 7]
```diff
@@ -6,7 +6,7 @@ def create
     user = User.find_by email: params[:session][:email].downcase
     if user && user.authenticate(params[:session][:password])
       log_in user
-      params[:session][:remember_me] == Settings.CHECKBOOK_TRUE ? 
+      params[:session][:remember_me] == Settings.CHECKBOOK_TRUE ?
```
Comment (dieunb) [inline app/controllers/sessions_controller.rb]: `CHECKBOOK` or `CHECKBOX`

**Comment by dieunb** [comment]
OK

**Pull request merged**